tests/ga/test_exthandlers_download_extension.py

```diff
@@ -35,6 +36,8 @@ def setUp(self):
         ext_handler.version = "1.0.0"
 
         protocol = WireProtocol("http://Microsoft.CPlat.Core.RunCommandLinux/foo-bar")
+        protocol.client.get_host_plugin = Mock()
+        protocol.client.get_artifact_request = Mock(return_value=(None, None))
```

Comment:
The mocks in this class have very internal knowledge of the code they are testing. They need rewriting, but for now I just defined a mock HGAP to account for the code I refactored.